udata/harvest/backends/dcat.py

```diff
@@ -329,19 +342,23 @@ def walk_graph(self, url: str, fmt: str) -> Generator[tuple[int, Graph], None, N
             response = self.post(url, data=data, headers={"Content-Type": "application/xml"})
             response.raise_for_status()
 
-            content = response.content
-            tree = ET.fromstring(content, parser=self.xml_parser)
-            if tree.tag == "{" + OWS_NAMESPACE + "}ExceptionReport":
-                raise ValueError(f"Failed to query CSW:\n{content}")
+            text = response.text
+            tree = self.saxon_proc.parse_xml(xml_text=text)
+            self.xpath_proc.set_context(xdm_item=tree)
 
-            search_results = tree.find("csw:SearchResults", {"csw": CSW_NAMESPACE})
-            if not search_results:
```

Comment:
For your information, following the https://github.com/opendatateam/udata/pull/3378/ PR, we now have this warning, that should disappear with this PR.
```
/srv/dev/lib/python3.11/site-packages/udata/harvest/backends/dcat.py:350: FutureWarning: Truth-testing of elements was a source of confusion and will always return True in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
  if not search_results:
```